packages/cli/test/unit/commands/db/revert.test.ts

```diff
@@ -74,6 +74,45 @@ test("don't revert the last migration if it had no down migration", async () =>
 	expect(dataSource.destroy).not.toHaveBeenCalled();
 });
 
+test('print migration name in error message is the migration has a name', async () => {
+	//
+	// ARRANGE
+	//
+	class TestMigration implements IrreversibleMigration {
+		name = 'Test Migration';
+
+		async up() {}
+	}
+
+	const connectionOptions = DbConfig.getConnectionOptions();
+	const migrations = [TestMigration];
+	// @ts-expect-error property is readonly
+	connectionOptions.migrations = migrations;
+	const dataSource = mock<DataSource>();
+	// @ts-expect-error property is readonly, and I can't pass them the `mock`
+	// because `mock` will mock the down method and thus defeat the purpose
+	// of this test, because the tested code will assume that the migration has a
+	// down method.
```

Comment:
Something like this? `mock<DataSource>({ migrations: migrations.map((M) => new M()) })`

Reply:
`mock` will mock recursively everything, including optional things. So this ends up adding a `down` method to the migration. Even when setting `deep: false`.

I haven't figured out how to stop that from happening except changing it later.

> because `mock` will mock the down method and thus defeat the purpose of this test, because the tested code will assume that the migration has a down method.